pkg/operator/vitesstopo/register.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 PlanetScale Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vitesstopo
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"vitess.io/vitess/go/vt/proto/topodata"
+	"vitess.io/vitess/go/vt/topo"
+
+	v2 "planetscale.dev/vitess-operator/pkg/apis/planetscale/v2"
+	"planetscale.dev/vitess-operator/pkg/operator/lockserver"
+	"planetscale.dev/vitess-operator/pkg/operator/results"
+)
+
+const (
+	// topoRequeueDelay is how long to wait before retrying when a topology
+	// server call failed. We typically return success with a requeue delay
+	// instead of returning an error, because it's unlikely that retrying
+	// immediately will be worthwhile.
+	topoRequeueDelay = 5 * time.Second
+)
+
+type RegisterCellsCmd struct {
+	Ctx context.Context
+	// EventObj holds the object type that the recorder will use when writing events.
+	EventObj         runtime.Object
+	Ts               *topo.Server
+	Recorder         *record.EventRecorder
```

Comment:
`EventRecorder` is an interface so it's already inherently a pointer (plus type info).

```suggestion
	Recorder         record.EventRecorder
```

Reply:
Good catch, will fix.

Reply:
Fixed https://github.com/planetscale/vitess-operator/pull/48/commits/7686cc38f6a7f91d4e0e57aa98d4cf90388d0721.